metricbeat/module/vsphere/cluster/cluster.go

```diff
@@ -166,9 +186,30 @@ func getAssetNames(ctx context.Context, pc *property.Collector, cl *mo.ClusterCo
 		}
 	}
 
-	return &assetNames{
+	return assetNames{
 		outputNetworkNames:   outputNetworkNames,
 		outputDatastoreNames: outputDatastoreNames,
 		outputHostNames:      outputHostNames,
 	}, nil
 }
+
+func getTriggerdAlarm(ctx context.Context, pc *property.Collector, triggeredAlarmState []types.AlarmState) ([]triggerdAlarm, error) {
+	var triggeredAlarms []triggerdAlarm
+	for _, alarmState := range triggeredAlarmState {
+		var triggeredAlarm triggerdAlarm
+		var alarm mo.Alarm
+		err := pc.RetrieveOne(ctx, alarmState.Alarm, nil, &alarm)
```

Comment:
Where is the alarm state being compared to red "critical alarms ?

Reply:
@ishleenk17 As we discussed yesterday, we will fetch all the triggered alarms  and ingest them, filtering will be done in pipeline of kibana.

Reply:
Are we able to access `triggered_alarms.state` in the triggered alarms `object`, if user wants to get only red alarms there?
Can you share here the processor
We will be able to do it in ingest pipeline in Integrations.
Are we mentioning somewhere which processors can they use to achieve the same result in beats ?

Reply:
Yes i have tried to access `triggered_alarm.state` and fetch red alarms.

In beats also Processors are supported which can be used to fetch red alarms. user can use something like [script processor](https://www.elastic.co/guide/en/beats/metricbeat/current/processor-script.html).